### PR TITLE
Added k8s test scenarios to ginkgo-based test framework

### PR DESCRIPTION
**Pull request by rksharma95**
* Added test scenarios to test `Kubearmor` in k8s env by applying `KubeArmorSecurityPolicy` for process, files and network operations.
* Added utility function `func K8sApplyFile(fileName string) error{}`
   supported k8s resources can be created with the YAML file if doesn't already exist.
   currently supported k8s resources:
   - Deployment
   - Service
   - Namespace
   - KubeArmorPolicy
   - KubeArmorHostPolicy   

- [x] validate if all the `local.sh` scenarios are covered.

Signed-off-by: rk <ramakant@accuknox.com>

### CODE REVIEW

**Comment by rksharma95** [comment]
PTAL @nyrahul

File: tests/util/kartutil.go [hunk 1]
```diff
@@ -297,6 +302,117 @@ func KspDeleteAll() {
...
+// K8sApplyFile can apply deployments, services, namespace, and kubearmorhostpolicy
+func K8sApplyFile(fileName string) error {
+	f, err := ioutil.ReadFile(fileName)
+
+	if err != nil {
+		return fmt.Errorf("error reading file %v\n", err.Error())
+
+	}
+	// register ksp scheme
+	kspScheme.AddToScheme(scheme.Scheme)
```
Comment (nyrahul) [inline tests/util/kartutil.go]: Please handle the return error here.

File: tests/util/kartutil.go [hunk 2]
```diff
@@ -297,6 +302,117 @@ func KspDeleteAll() {
...
+
+	// register hsp scheme
+	hspScheme.AddToScheme(scheme.Scheme)
```
Comment (nyrahul) [inline tests/util/kartutil.go]: Please handle the return error here.

File: tests/util/kartutil.go [hunk 3]
```diff
@@ -297,6 +302,117 @@ func KspDeleteAll() {
 	}
 }
 
+// K8sApplyFile can apply deployments, services, namespace, and kubearmorhostpolicy
+func K8sApplyFile(fileName string) error {
+	f, err := ioutil.ReadFile(fileName)
+
+	if err != nil {
+		return fmt.Errorf("error reading file %v\n", err.Error())
+
```
Comment (nyrahul) [inline tests/util/kartutil.go]:
```suggestion
		return log.Errorf("error reading file %v", err.Error())
```

File: tests/util/kartutil.go [hunk 4]
```diff
@@ -297,6 +302,117 @@ func KspDeleteAll() {
...
+	for _, f := range sepYamlfiles {
+		if f == "\n" || f == "" {
+			// ignore empty cases
+			continue
+		}
+
+		decode := scheme.Codecs.UniversalDeserializer().Decode
+		obj, _, err := decode([]byte(f), nil, nil)
+		if err != nil {
+			fmt.Printf("%v", err)
```
Comment (nyrahul) [inline tests/util/kartutil.go]: Please use `logrus` log functions for error/general printing.

File: tests/util/kartutil.go [hunk 5]
```diff
@@ -297,6 +302,117 @@ func KspDeleteAll() {
...
+			namespace := deployment.Namespace
+
+			result, err := k8sClient.K8sClientset.AppsV1().Deployments(namespace).Create(context.TODO(), deployment, metav1.CreateOptions{})
+			if err != nil {
+				if !strings.Contains(err.Error(), "already exists") {
+					return err
+				} else {
+					fmt.Printf("Deployment %s already exists ...\n", deployment.Name)
+					continue
+				}
```
Comment (nyrahul) [inline tests/util/kartutil.go]:
```suggestion
				if strings.Contains(err.Error(), "already exists") {
              			        fmt.Printf("Deployment %s already exists ...\n", deployment.Name)
				        continue
				}
				return err
```

File: tests/util/kartutil.go [hunk 6]
```diff
@@ -297,6 +302,117 @@ func KspDeleteAll() {
...
+					return err
+				} else {
+					fmt.Printf("Namespace %s already exists ...\n", ns.Name)
+					continue
+				}
+			}
+			fmt.Printf("Namespace %s created ...\n", ns.Name)
+		}
+	}
+	time.Sleep(1 * time.Second)
```
Comment (nyrahul) [inline tests/util/kartutil.go]:
```suggestion
```

**Review by nyrahul — DISMISSED** [review]

File: tests/util/kartutil.go (hunk 6 above)
Comment (rksharma95) [reply tests/util/kartutil.go]: on my system, the tests frequently failed, and this timeout here handled that case.

Comment (nyrahul) [reply tests/util/kartutil.go]: Why are the tests failed without this timeout?

Comment (rksharma95) [reply tests/util/kartutil.go]: in the absence of this delay, the behaviour was similar to the case when the policy has not been applied.

Comment (nyrahul) [reply tests/util/kartutil.go]: why?

**Comment by nyrahul** [comment]
Hey @rksharma95 , any updates here? Thanks

**Comment by nyrahul** [comment]
@rksharma95 , the tests are failing with the new apply function! Can we make it work?

**Review by achrefbensaad — CHANGES_REQUESTED** [review]
Binaries should'nt be included in PR's.
KubeArmor/BPF/tests/checks/security_path_unlink.bpf.o
KubeArmor/BPF/tests/syscheck

**Comment by achrefbensaad** [comment]
LGTM !!!

**Review by achrefbensaad — APPROVED** [review]

**Pull request merged**